### PR TITLE
scheduler: add grant hot region scheduler for test imbalance scene 

### LINKED ISSUE #4250: PD should offer new scheduler to imbalance hot region for test stablity
## Feature Request
I find the init bench environment is different because of scheduler in hot region balance region test, I want that the bench enviroment should be save in every test .

My test steps is below:
1. add shuffle region scheduler and shuffle leader region scheduler in the early 20 minutes 
2.  run tpcc  bench with closing hot region scheduler 
3.  run tppc bench again with hot region scheduler  
### Describe your feature request related problem

image

### Describe the feature you'd like
 I hope that PD should offer new scheduler named grant_hot_region to place  hot region the fixed three tikvs. 

### Describe alternatives you've considered
tikv hot region distribution is similar in  the new scheduler

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?
add grant hot region scheduler for test hot region imbalance scene, close #4250

### What is changed and how it works?
It will move hot region to the fixed stores and transfer leader to one store 
### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

image

Code changes

Side effects

Related changes

### Release note

```release-note
none
```

### CODE REVIEW

File: server/api/scheduler.go [hunk 1]
```diff
@@ -198,6 +198,13 @@ func (h *schedulerHandler) Post(w http.ResponseWriter, r *http.Request) {
 			h.r.JSON(w, http.StatusInternalServerError, err.Error())
 			return
 		}
+	case schedulers.GrantHotRegionName:
+		leaderID := input["store-leader-id"].(string)
```
Comment (lhy1024) [inline server/api/scheduler.go:202]: use collectStringOption?

File: server/schedulers/grant_hot_region.go [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+	return router
+}
+
+func (s *grantHotRegionScheduler) Schedule(cluster opt.Cluster) []*operator.Operator {
+	schedulerCounter.WithLabelValues(s.GetName(), "schedule").Inc()
+	i := s.r.Int() % len(s.types)
+	return s.dispatch(s.types[i], cluster)
+}
+
+func (s *grantHotRegionScheduler) dispatch(typ rwType, cluster opt.Cluster) []*operator.Operator {
```
Comment (lhy1024) [inline server/schedulers/grant_hot_region.go:263]: Will it conflict with other schedulers?

Comment (bufferflies) [reply server/schedulers/grant_hot_region.go:263]: yes,it will conflict with balance hot region scheduler

**Review by disksing — APPROVED** [review]

**Comment by nolouch** [comment]
need a relative issue.

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
@bufferflies plz fix the `invalid-commit-message`

File: server/api/scheduler.go (hunk 1 above)
Comment (bufferflies) [reply server/api/scheduler.go:202]: yes, it will add param validate.

**Pull request merged**